To allow MP to begin dates up to 90 days in future. (fix/#6534-allow-future-begin-dates)

spetros-do opened the pull request:
### Changes Made:
- Updated backend MAXIMUM_FUTURE_DATE constant from 30 to 90 days
- Modified frontend date picker to allow future date selection up to 90 days for:
    - Defaults
    - Formulas
    - Loads
    - Location Attributes
    - Methods
    - Qualifications
    - Rectangular Duct WAFs
    - Spans
    - Systems
    - Unit Information

### Testing:
- Verify users can select and save begin dates up to 90 days in the future on all specified screens.


```

EvanSun220 committed:
feature/#6617 update the payload which send to the endpoint that update em submission record

Commit:
Remove Calculated Values and Report Errors Columns from Global Views

ergjustin committed:
Merge branch 'develop' into fix/#6609_remove_calculate_values_report_errors

Commit:
in queue with place number

spetros-do committed:
To allow MP begin dates up to 90 days in future

spetros-do committed:
Merge branch 'develop' into fix/#6534-allow-future-begin-dates

spetros-do commented:
https://github.com/US-EPA-CAMD/easey-ui/issues/6534

EvanSun220 committed:
Merge branch 'develop' into feature/#6617-EM-Submission-Access-Windows-Marked-as-Delete-Should-Display-in-UI

mark-hayward-erg committed:
Merge pull request #1558 from US-EPA-CAMD/feature/#6617-EM-Submission-Access-Windows-Marked-as-Delete-Should-Display-in-UI

feature/#6617 EM Submission Access Windows Marked as Delete Should Display in UI

Commit:
remove MATS import option from QA Test

Commit:
remove matsBulkFilesRef

Commit:
Merge branch 'develop' of https://github.com/US-EPA-CAMD/easey-ecmps-ui into fix/#6609_remove_calculate_values_report_errors

Commit:
Merge branch 'fix/#6609_remove_calculate_values_report_errors' of https://github.com/US-EPA-CAMD/easey-ecmps-ui into fix/#6609_remove_calculate_values_report_errors

ergjustin committed:
Merge pull request #1564 from US-EPA-CAMD/fix/#6514_Remove_Mats_import_QA_Test_2

remove MATS import option from QA Test

Jhernandez29 committed:
Merge branch 'develop' into fix/#6609_remove_calculate_values_report_errors

ergjustin committed:
Merge pull request #1559 from US-EPA-CAMD/fix/#6609_remove_calculate_values_report_errors

Remove Calculated Values and Report Errors Columns from Global Views

dependabot[bot] committed:
build(deps-dev): bump vite from 6.2.4 to 6.2.5

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 6.2.4 to 6.2.5.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v6.2.5/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.2.5/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 6.2.5
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

mark-hayward-erg committed:
Merge pull request #1565 from US-EPA-CAMD/dependabot/npm_and_yarn/vite-6.2.5

build(deps-dev): bump vite from 6.2.4 to 6.2.5

ergjustin committed:
Merge branch 'develop' into feat/#6619_in_queue_with_place

djw4erg committed:
Merge pull request #1560 from US-EPA-CAMD/feat/#6619_in_queue_with_place

in queue with place number

ergjustin commented on src/components/ModalDetails/ModalDetails.js:
```diff
@@ -376,11 +376,20 @@ const ModalDetails = ({
-            }
-          />
+                  // Allow future dates up to 90 days for specific components
+                  ["Defaults", "Formulas", "Loads", "Location Attributes", "Methods",
+                    "Qualifications", "Rectangular Duct WAFs", "Spans", "Systems",
+                    "Unit Information"].includes(title)
```
Could you please remove the extra 's' from the following title list?

- Defaults → Default
- Formulas → Formula
- Loads → Load
- Location Attributes → Location Attribute
- Methods → Method
- Qualifications → Qualification
- Rectangular Duct WAFs → Rectangular Duct WAF
- Spans → Span
- Systems → System
- Create Systems → Create System

Also, I’m not sure if all Unit information needs to be included—could you kindly confirm? If needed, please ensure the array list includes: 

- Unit Fuel
-  Unit Control
- Unit Capacity

spetros-do replied:
- Thanks for catching the array list count and plural format. 
- The update is not just for Protocol Gas; in the original code, it had a special case for "Protocol Gas", where maxDate was set to null -- meaning no date restriction; while all other components had a maxDate set to the current date. 
- Based on the requirement described on the change made maintains the special case for "Protocol Gas"; but adds a new condition for the components that should allow users to select dates up to 90 days in the future.

dependabot[bot] committed:
build(deps-dev): bump vite from 6.2.5 to 6.2.6

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 6.2.5 to 6.2.6.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v6.2.6/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.2.6/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 6.2.6
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

mark-hayward-erg committed:
Merge pull request #1568 from US-EPA-CAMD/dependabot/npm_and_yarn/vite-6.2.6

build(deps-dev): bump vite from 6.2.5 to 6.2.6

spetros-do committed:
Updated code to allow MP to begin dates up to 90 days in the future.

spetros-do committed:
Merge branch 'fix/#6534-allow-future-begin-dates' of https://github.com/US-EPA-CAMD/easey-ecmps-ui into fix/#6534-allow-future-begin-dates

# Conflicts:
#	src/components/ModalDetails/ModalDetails.js

spetros-do commented:
Replaced by PR: https://github.com/US-EPA-CAMD/easey-ecmps-ui/pull/1571
